week02/Implement_strStr.cpp

```diff
@@ -0,0 +1,21 @@
+//Problem 5: Implement strStr()
+
+//Solution:
+
+//Time Complexity: O(N) where N is the length of haystack string.
+//Space Complexity: O(1)
+
+int strStr(string haystack, string needle){
+    if(needle == "") return 0;
+    else {
+        int haystackLength = haystack.length();
+        int needleLength = needle.length();
+        for(int i = 0; i <= (haystackLength - needleLength); i++){
+            if(haystack[i] == needle[0] &&
+                haystack.substr(i, needleLength) == needle){
+                return i;
+            }
+        }
+        return -1;
+    }
+}
```

Comment:
```suggestion
int strStr(string haystack, string needle){
        for(int i = 0; i <= (haystack.length()- needle.length()); i++){
            if(haystack.substr(i, needleLength) == needle){
                return i;
            }
        }
        return -1;
}
```

It does the same work, but check the readability and concise-ness.